File: jscomp/test/obj_magic_test.js
'use strict';

var Mt = require("./mt.js");

function is_block(x) {
  return typeof x !== "number";
}

var suites_0 = [
  "is_block_test1",
  (function (param) {
      return {
              TAG: /* Eq */0,
              _0: false,
              _1: false
            };
    })
];

var suites_1 = {
  hd: [
    "is_block_test2",
    (function (param) {
        return {
                TAG: /* Eq */0,
                _0: true,
                _1: typeof ({
                    hd: 3,
                    tl: /* [] */0
                  }) !== "number"
              };
      })
  ],
  tl: {
    hd: [
      "is_block_test3",
      (function (param) {
          return {
                  TAG: /* Eq */0,
                  _0: true,
                  _1: true
                };
        })
    ],
    tl: {
      hd: [
        "is_block_test4",
        (function (param) {
            return {
                    TAG: /* Eq */0,
                    _0: false,
                    _1: false
                  };
          })
      ],
      tl: /* [] */0
    }
  }
};

var suites = {
  hd: suites_0,
  tl: suites_1
};

Mt.from_pair_suites("Obj_magic_test", suites);

exports.is_block = is_block;
exports.suites = suites;
/*  Not a pure module */

File: jscomp/test/obj_magic_test.ml
[@@@bs.config {
  flags = [|
    "-drawlambda"
  |]
}]


(* let empty_backtrace  = Obj.obj (Obj.new_block Obj.abstract_tag 0) *)

let is_block x = (Js.typeof  (Obj.repr x) <> "number")
 
let suites = Mt.[
"is_block_test1", (fun _ -> Eq (false, is_block 3));
"is_block_test2", (fun _ -> Eq (true, is_block [3]));
"is_block_test3", (fun _ -> Eq(true, is_block "x"));
"is_block_test4", (fun _ -> Eq (false, is_block 3.0))
]

;; Mt.from_pair_suites __MODULE__ suites
